go/vt/vtgate/semantics/binder.go

```diff
@@ -98,6 +136,44 @@ func (b *binder) up(cursor *sqlparser.Cursor) error {
 	return nil
 }
 
+func (b *binder) rewriteJoinUsingColName(deps dependency, node *sqlparser.ColName, currentScope *scope) (dependency, error) {
+	newTbl := deps.recursive.Constituents()[0]
```

Comment:
nit: it's probably safer to first check if the constituents slice is not empty or nil